packages/extension/src/view/devtools/components/privateAdvertising/protectedAudience/explorableExplanation/auctionEventTransformers.ts

```diff
@@ -196,72 +208,172 @@ export const createAuctionEvents = (
   seller: string,
   advertisers: string[],
   eventStartTime: number,
-  isTopLevelBid: boolean
+  isTopLevelBid: boolean,
+  currentStep: StepType | null,
+  previousEvents: singleAuctionEvent[] | null,
+  isMultiSeller = false
 ) => {
   const bidEvents = transformBidEvent(interestGroups, isTopLevelBid);
 
   bidEvents.sort((a, b) => (a?.bid ?? 0) - (b?.bid ?? 0));
-
-  const events = [
-    transformStartedEvent(seller),
-    transformConfigResolvedEvent(seller),
-    transformLoadedEvent(interestGroups),
-    transformFetchingEvents(
-      advertisers,
-      'start',
-      seller,
-      'bidderTrustedSignals'
-    ),
-    transformFetchingEvents(advertisers, 'start', seller, 'bidderJS'),
-    transformFetchingEvents(advertisers, 'start', seller, 'sellerJS'),
-    transformFetchingEvents(advertisers, 'finish', seller, 'bidderJS'),
-    transformFetchingEvents(
-      advertisers,
-      'start',
-      seller,
-      'sellerTrustedSignals'
-    ),
-    bidEvents,
-    transformFetchingEvents(
-      advertisers,
-      'finish',
-      seller,
-      'sellerTrustedSignals'
-    ),
-    {
-      formattedTime: '',
-      uniqueAuctionId: '27A93A016A30D0A5FB7B8C8779D98AF8',
-      name: bidEvents[bidEvents.length - 1].name,
-      ownerOrigin: bidEvents[bidEvents.length - 1].ownerOrigin,
-      type: 'win',
-      time: 1734076670.129756,
-      bid: bidEvents[bidEvents.length - 1].bid,
-      bidCurrency: bidEvents[bidEvents.length - 1].bidCurrency,
-      eventType: 'interestGroupAccessed' as singleAuctionEvent['eventType'],
-    },
-    transformFetchingEvents(
-      advertisers,
-      'finish',
-      seller,
-      'bidderTrustedSignals'
-    ),
-  ];
+  const events: singleAuctionEvent[] = [...(previousEvents ?? [])];
+
+  switch (currentStep?.title) {
+    case 'runAdAuction()':
+      events.push(
+        transformStartedEvent(seller),
+        transformConfigResolvedEvent(seller)
+      );
+      break;
+    case 'Load Interest Group':
```

Comment:
We should avoid checking against text string, instead let's create some key value pair object variable/constant, so that if the text changes, we don't have to update here.

Reply:
In fact we already have it in `flowConfig.jsx`.

Reply:
Resolved in latest commits.